### PR TITLE
Try letting NSDictionary duck-typecheck things instead of doing it ourselves the slow way

### PR DESCRIPTION
**Pull request by Catfish-Man**
Fixes https://bugs.swift.org/browse/SR-11154

Fixes rdar://problem/53229663

### CODE REVIEW

**Comment by Catfish-Man** [comment]
@swift-ci please test

**Comment by Catfish-Man** [comment]
@swift-ci please benchmark

File: stdlib/public/Darwin/Foundation/NSDictionary.swift [hunk 1]
```diff
@@ -13,6 +13,13 @@
 @_exported import Foundation // Clang module
 import _SwiftFoundationOverlayShims
 
+// We don't check for NSCopying here for performance reasons. We would
+// just crash anyway, and NSMutableDictionary will still do that when
+// it tries to call -copyWithZone: and it's not there
+func duckCastToNSCopying(_ x: Any) -> NSCopying {
```
Comment (xwu) [inline stdlib/public/Darwin/Foundation/NSDictionary.swift]:
```suggestion
private func duckCastToNSCopying(_ x: Any) -> NSCopying {
```

Any reason it needs to be internal?

Comment (Catfish-Man) [reply stdlib/public/Darwin/Foundation/NSDictionary.swift]: Nope, good catch

**Comment by Catfish-Man** [comment]
@swift-ci please smoke test and merge

**Comment by Catfish-Man** [comment]
@swift-ci please smoke test and merge

**Comment by shahmishal** [comment]
@swift-ci please smoke test and merge

**Pull request merged**